search/connection.py

```diff
@@ -93,15 +93,22 @@ def get_active_aliases(conn, object_types):
     """
     if not object_types:
         object_types = VALID_OBJECT_TYPES
-    return [
-        alias
-        for alias_tuple in [
-            (get_default_alias_name(obj), get_reindexing_alias_name(obj))
-            for obj in object_types
+    if update_only:
+        return [
+            alias
+            for alias in [get_default_alias_name(obj) for obj in object_types]
+            if conn.indices.exists(alias)
+        ]
+    else:
+        return [
+            alias
+            for alias_tuple in [
+                (get_default_alias_name(obj), get_reindexing_alias_name(obj))
+                for obj in object_types
+            ]
+            for alias in alias_tuple
+            if conn.indices.exists(alias)
         ]
```

Comment:
This function is getting kind of complex now so I think it'd be worth pulling apart into some separate functions and composing them together:
```suggestion
    aliases = []
    for obj_type in object_types:
        aliases.append(get_default_alias_name)
        if include_reindexing:
            aliases.append(get_reindexing_alias_name(obj_type))
    return [alias for alias in alias if conn.alias.exists(alias)]
```